src/plugins/Sharesite/Freesite.java

```diff
@@ -90,6 +111,35 @@ public synchronized void setName(String name) {
 		this.name = name;
 	}
 
+	public synchronized String getPath() {
+		if (path != null) {
+			return path;
+		} else {
+			return name;
+		}
+	}
+
+	public synchronized void setPath(String path) {
+		this.path = path;
+	}
+
+	public synchronized Integer getInsertHour() {
+		if (insertHour != null) {
+			return insertHour;
+		} else {
+			return -1; // meaning unrestricted
+		}
+	}
+
+	public synchronized void setInsertHour(Integer insertHour) {
+		if (insertHour.equals(-2)) {
```

Comment:
Oh, yes, it’s that easy. I keep forgetting that java is pretty low-level on this (not meant as derogative: it feels high level most of the time but can suddenly drop very low and allows writing efficient code on a pretty low level — something I would not dare to do in Python).

insertHour == null could happen if there’s something wrong with the storage.